### ISSUE TITLE
`rosdep resolve` on version 0.16.2 yields `TypeError`

### ISSUE THREAD

**Issue by acarrillo** [ros-infrastructure/rosdep#720]
I installed `rosdep` from PyPI. `rosdep resolve` yields the failing the stack trace below:
```
acarrillo ~ $ rosdep resolve

ERROR: Rosdep experienced an error: argument of type 'NoneType' is not iterable
Please go to the rosdep page [1] and file a bug report with the stack trace below.
[1] : http://www.ros.org/wiki/rosdep

rosdep version: 0.16.2

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 144, in rosdep_main
    exit_code = _rosdep_main(args)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 406, in _rosdep_main
    if os.environ['ROS_DISTRO'] in python_versions:
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by acarrillo** [comment]
I'm using ROS kinetic on Ubuntu 16.

**Comment by sloretz** [comment]
Does the error persist after running `rosdep update`?

**Comment by acarrillo** [comment]
@sloretz thanks for checking in, I am unable to `rosdep update` as well:

```
acarrillo ~ $ rosdep update

ERROR: Rosdep experienced an error: argument of type 'NoneType' is not iterable
Please go to the rosdep page [1] and file a bug report with the stack trace below.
[1] : http://www.ros.org/wiki/rosdep

rosdep version: 0.16.2

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 144, in rosdep_main
    exit_code = _rosdep_main(args)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 406, in _rosdep_main
    if os.environ['ROS_DISTRO'] in python_versions:
TypeError: argument of type 'NoneType' is not iterable

```

**Comment by sloretz** [comment]
Hmm, does it persist after doing `ROS_PYTHON_VERSION=2 rosdep update`?

**Comment by acarrillo** [comment]
Nope that fixes it -- if I'm reading the intent of #708, that PR was supposed to set that variable if it doesn't exist ( as is the case on my + my CI machines); is some other ROS setup software supposed to be setting `ROS_PYTHON_VERSION`?
